models/forward_tacotron.py

```diff
@@ -103,69 +116,95 @@ def forward(self, x):
                                           conv_dims=pitch_conv_dims,
                                           rnn_dims=pitch_rnn_dims,
                                           dropout=pitch_dropout)
+        self.energy_pred = SeriesPredictor(num_chars=num_chars,
+                                           emb_dim=series_embed_dims,
+                                           conv_dims=energy_conv_dims,
+                                           rnn_dims=energy_rnn_dims,
+                                           dropout=energy_dropout)
         self.prenet = CBHG(K=prenet_k,
                            in_channels=embed_dims,
                            channels=prenet_dims,
                            proj_channels=[prenet_dims, embed_dims],
-                           num_highways=highways)
-        self.lstm = nn.LSTM(2 * prenet_dims + pitch_emb_dims,
-                            rnn_dim,
+                           num_highways=num_highways)
+        self.lstm = nn.LSTM(2 * prenet_dims + pitch_emb_dims + energy_emb_dims,
+                            rnn_dims,
                             batch_first=True,
                             bidirectional=True)
-        self.lin = torch.nn.Linear(2 * rnn_dim, n_mels)
+        self.lin = torch.nn.Linear(2 * rnn_dims, n_mels)
         self.register_buffer('step', torch.zeros(1, dtype=torch.long))
         self.postnet = CBHG(K=postnet_k,
                             in_channels=n_mels,
                             channels=postnet_dims,
                             proj_channels=[postnet_dims, n_mels],
-                            num_highways=highways)
+                            num_highways=num_highways)
         self.dropout = dropout
         self.post_proj = nn.Linear(2 * postnet_dims, n_mels, bias=False)
         self.pitch_emb_dims = pitch_emb_dims
+        self.energy_emb_dims = energy_emb_dims
         if pitch_emb_dims > 0:
             self.pitch_proj = nn.Sequential(
                 nn.Conv1d(1, pitch_emb_dims, kernel_size=3, padding=1),
                 nn.Dropout(pitch_proj_dropout))
+        if energy_emb_dims > 0:
+            self.energy_proj = nn.Sequential(
+                nn.Conv1d(1, energy_emb_dims, kernel_size=3, padding=1),
+                nn.Dropout(energy_proj_dropout))
+
+    def forward(self, batch: Dict[str, torch.tensor]) -> Dict[str, torch.tensor]:
+        x = batch['x']
+        mel = batch['mel']
+        dur = batch['dur']
+        mel_lens = batch['mel_len']
+        x_lens = batch['x_len'].cpu()
+        pitch = batch['pitch'].unsqueeze(1)
+        energy = batch['energy'].unsqueeze(1)
 
-    def forward(self, x, mel, dur, mel_lens, pitch):
         if self.training:
             self.step += 1
 
-        x = self.embedding(x)
-        dur_hat = self.dur_pred(x).squeeze()
-        pitch_hat = self.pitch_pred(x).transpose(1, 2)
-        pitch = pitch.unsqueeze(1)
+        dur_hat = self.dur_pred(x, x_lens=x_lens).squeeze(-1)
```

Comment:
just checking: I understood that the embeddings were already independent before this PR. It would seem that they are only now